### ISSUE TITLE
Deal with long filenames

### ISSUE THREAD

**Issue by casey** [agora-org/agora#107]
Currently, we store filenames in the invoice memo field. The memo field is limited to 639, so if we have longer paths, this will fail.

**Comment by anantasty** [comment]
What is the current best paths to resolve this ? 
I would be happy to push a fix if you set me up with the right context @casey

**Comment by casey** [comment]
This is a tricky one, and I'm not sure this is a good issue to work on right away. We don't have a database, so right now we're using LND to store data related to generated invoice. In particular, we store the file that an invoice is for in the invoice memo field, and retrieve it from LND to figure out which file to serve for a paid invoice. That field is limited to 639 characters, but to store more data, we'd need to add an agora-specific data-store, like SQLite, so we could put state there. We *probably* want to do that, but it's not a high priority at the moment, so this issue should probably be deferred until someone actually needs longer file names, or we have a data store so it becomes easy.

**Comment by anantasty** [comment]
Makes sense 639 characters is long enough for a large majority of use cases
